### DISCUSSION TITLE
Trying to use your macros.cfg, but every time i get Move exceeds maximum extrusion (1.011mm^2 vs 0.640mm^2). I use CURA. What should i do? Thank you!

### DISCUSSION THREAD

**Discussion by LeeOtts** [LeeOtts/Ender3v2-Klipper-Configs#13] [Q&A]
Trying to use your macros.cfg, but every time i get Move exceeds maximum extrusion (1.011mm^2 vs 0.640mm^2). I use CURA. What should i do? Thank you!

_Originally posted by @master-of-the-dungeon in https://github.com/LeeOtts/Ender3v2-Klipper-Configs/discussions/6#discussioncomment-7868203_

**Comment by LeeOtts** [comment]
@master-of-the-dungeon

You can add this to your extruder section in the printer config. 

max_extrude_cross_section: 5

**Comment by tbastosc** [comment]
hello, to avoid pass over to negative positions, I edited Line_Purge.cfg, macro. 

`
gcode:

    {% if display_parameters == True %}
      { action_respond_info("adaptive_enable : %d" % (adaptive_enable))  }
      { action_respond_info("z_height        : %f" % (z_height))  }
      { action_respond_info("purge_amount    : %f" % (purge_amount))  }
      { action_respond_info("line_length     : %f" % (line_length))  }
      { action_respond_info("flow_rate       : %f" % (flow_rate))  }
      { action_respond_info("x_default       : %f" % (x_default))  }
      { action_respond_info("y_default       : %f" % (y_default))  }
      { action_respond_info("distance_to_object_y : %f" % (distance_to_object_y))  }
    {% endif %}

    {% if adaptive_enable == True %}
        {% set all_points = printer.exclude_object.objects | map(attribute='polygon') | sum(start=[]) %}
        {% set x_origin = (all_points | map(attribute=0) | min | default(x_default)) %}
        {% set y_origin = (all_points | map(attribute=1) | min | default(y_default)) %}
        {% set x_origin = ([x_origin, 0] | max) %}
        {% set y_origin = ([y_origin, 0] | max) %}
    {% else %}
        {% set x_origin = x_default | float %}
        {% set y_origin = y_default | float %}
    {% endif %}
    {% set nozzle_dia = printer.configfile.config.extruder.nozzle_diameter | float %}
    {% set cross_section = nozzle_dia * z_height | float %}
    {% set purge_move_speed = (cross_section * flow_rate) * 60 | float %}
    {% set travel_speed = (printer.toolhead.max_velocity) * 60 | float %}

    # Garantir que a posição de purga em Y não é negativa
    {% set y_purge_position = ([y_origin - distance_to_object_y, 0] | max)+ 3 %} #>>>>>>>>>>avoid go to negative positions + y 3mm <<<<<<<

    G92 E0                                                                              # Reset extruder
    G0 F{travel_speed}                                                                  # Set travel speed
    G90                                                                                 # Absolute positioning
    #G0 X{x_origin} Y{y_origin - distance_to_object_y}  |||||||||||||||||||||||||||||||||||||||||||||||alteraçao 1
    G0 X{x_origin} Y{y_purge_position}  #>>>>>>>>>>>>>> Move to safe Y position <<<<<<<<<<<<<<<<
    G0 Z{z_height}                                                                      # Move to purge Z height
    M83                                                                                 # Relative extrusion mode
    G1 X{x_origin + line_length} E{purge_amount} F{purge_move_speed}                    # Purge line
    G1 E-.5 F2100                                                                       # Retract
    G92 E0                                                                              # Reset extruder distance
    M82                                                                                 # Absolute extrusion mode
    G0 Z{z_height * 2} F{travel_speed}                                                  # Z hop`
